pkg/services/alerting/notifier.go

```diff
@@ -119,7 +123,7 @@ func (n *notificationService) sendNotifications(evalContext *EvalContext, notifi
 }
 
 func (n *notificationService) uploadImage(context *EvalContext) (err error) {
-	uploader, err := imguploader.NewImageUploader()
+	uploader, err := newImageUploaderProvider()
```

Comment:
It seems it is to make it replaceable during testing? With that being the case - I feel like somehow we should need need make something like this just for testing. But I can't think of anything better right now - maybe there isn't. 

So if this is only wrapped so it can be replaced for tests - I guess just add a comment to explain that.

Reply:
I'm wondering the same.

Reply:
Yeah a classic go trick for stubbing out things in tests. Will add a comment.